feat(send-data): improve OpenTelemetry logs collection configurations (#2501)
SumoLogic/sumologic-documentation: docs/send-data/opentelemetry-collector/data-source-configurations/collect-logs.md

---

The Sumo Logic Distribution for OpenTelemetry Collector provides various receivers for log collection. This document describes the most common scenarios.

* [Collecting logs from local files](#collecting-logs-from-local-files)
  * [Parsing JSON logs](#parsing-json-logs)
* [Collecting logs from Windows Event Log](#collecting-logs-from-windows-event-log)
* [Collecting logs from Syslog](#collecting-logs-from-syslog)
  * [Parsing Syslog logs into structured logs](#parsing-syslog-logs-into-structured-logs)
  * [Collecting Syslog logs in format compatible with Sumo Logic Installed Collector](#collecting-syslog-logs-in-format-compatible-with-sumo-logic-installed-collector)

You can find the full list of receivers on our [Sumo Logic OpenTelemetry Collector page](https://github.com/SumoLogic/sumologic-otel-collector/tree/main#components).

:::tip Additional information
See [Additional Configurations Reference](/docs/send-data/opentelemetry-collector/data-source-configurations/additional-configurations-reference/) for more details about configuring the Sumo Logic OpenTelemetry Collector.
:::

## Collecting logs from local files

Following configuration demonstrates:

> 1. **Collect**: How to collect logs stored in /var/log/syslog and /var/log/audit/audit.log

Reviewer: ```suggestion
1. **Collect**: How to collect logs stored in `/var/log/syslog` and `/var/log/audit/audit.log`
```

description: Learn how to collect logs using the Sumo Logic OpenTelemetry Collector.
---

The Sumo Logic Distribution for OpenTelemetry Collector provides various receivers for log collection. This document describes the receivers most commonly used for logs:

* [Filelog Receiver](#filelog-receiver)
* [Windows Log Event Receiver](#windows-log-event-receiver)
* [Syslog Receiver](#syslog-receiver)

> You can find the full list of receivers on our [Sumologic OpenTelemetry Collector page](https://github.com/SumoLogic/sumologic-otel-collector/tree/main#components).

Reviewer 2: ```suggestion
You can find the full list of receivers on our [Sumo Logic OpenTelemetry Collector page](https://github.com/SumoLogic/sumologic-otel-collector/tree/main#components).
```

## Filelog Receiver

> Following configuration demonstrates : 
>  1. Collect   : How to collect logs stored in /var/log/syslog and /var/log/audit/audit.log
>  2. Transform : Set \_sourceCatetgory field to "application_logs_prod"
>  3. Export    : Send data to authenticated Sumologic organization.

Reviewer 2: ```suggestion
Following configuration demonstrates: 
1. Collect. How to collect logs stored in /var/log/syslog and /var/log/audit/audit.log.
2. Transform. Set `_sourceCatetgory` field to `application_logs_prod`.
3. Export. Send data to authenticated Sumo Logic organization.
```

```yaml
receivers:

>   filelog/custom_files:                                         

Reviewer 3: The trailing spaces should be removed.

>     include_file_path: true

Reviewer 3: Why changing this from `include_file_path_resolved: true` to `include_file_path: true`?
Why is the `include_file_name: false` being removed?

    include:                                         
      - /var/log/syslog
      - /var/log/audit/audit.log
     storage: file_storage
     
processors:

>   resource/processor1:

Reviewer 3: Do we want to use this name `processor1`? I'd prefer something more descriptive.

    attributes:                                       
      - key: _sourceCategory
        value: application_logs_prod
        action: insert
service:
  pipelines:

>     logs/pipeline1:

Reviewer 3: `pipeline` - Is it a name we want to use? Perhaps we want to use something more descriptive?

      receivers:

>         - filelog/1

Reviewer 3: The name `filelog/1` is different from the `filelog/custom_files` defined above. The collector will fail to start with this config.

      processors:

>         - resource/1

Reviewer 3: The name `resource/1` is different from the above defined `resource/processor`.

        - batch

>         - memory_limiter

Reviewer 3: Memory limiter should be the first processor in the processors pipeline.

      exporters:
        - sumologic
```

### Configuration details

>  1. Create a file in folder /etc/otelcol-sumo/conf.d  with name for your choice, sample_app_configuration.yaml.
>  2. Paste the above content.
>  3. Restart collector with following command:
>       - Linux : systemctl restart otelcol-sumo
>       - Windows : Restart-Service -Name OtelcolSumo

Reviewer 2: ```suggestion
1. Create a file in folder /etc/otelcol-sumo/conf.d  with name for your choice, sample_app_configuration.yaml.
2. Paste the above content.
3. Restart collector with following command:
   - For Linux:
     ```
     systemctl restart otelcol-sumo
     ```
   - Windows:
     ```
     Restart-Service -Name OtelcolSumo
     ```
```

* __receivers__: 
  * `filelog/custom_files:` Unique name defined for the filelog receiver. You can add multiple file receivers as filelog/2 , filelog/prod etc.
  * `include:` Lets the Filelog Receiver know where the log file is placed. Make sure that the collector has permissions to access the files; otherwise, it will not be collected.
  * `include_file_path_resolved: true` Adds a `log.file.path_resolved` attribute to the logs that contain the whole path of the file.
  * `storage: file_storage:` Prevents the receiver from reading the same data over and over again on each otelcol restart. Extension defined by default in sumologi.yaml

>  
> * __processors__: 
>   * `resource/1` Name of the processor that uses `attributes` to add \_sourceCategory field with value `application_logs_prod`
>  
> * __exporters__: 
>   *  `sumologic` Sends data to registered Sumologic Organization.  Auto configured in sumologic.yaml during installation using script.
>  

Reviewer 2: ```suggestion
* __processors__: 
  * `resource/1` Name of the processor that uses `attributes` to add `_sourceCategory` field with value `application_logs_prod`.
* __exporters__: 
  *  `sumologic` Sends data to registered Sumologic Organization.  Auto-configured in sumologic.yaml during installation using script.
```

* __service__: 
  *  `logs/pipeline1:`  Pipeline glues together what data to collect(receiver), how to process the data(processors) and where to send logs(exporter).

> 

Reviewer 2: ```suggestion
```

The remaining processors in pipeline are from `sumologic.yaml` file and should be applied for better performance of the collector.

:::note
The receiver (`filelog/custom_files`) ,  (`resource/processor1`) and  pipeline (`logs/pipeline1:`) names should be unique across all configuration files to avoid conflicts and unexpected behavior.
:::

For more details, see the [Filelog Receiver documentation][filelogreceiver_readme].

## Windows Log Event Receiver

Windows Log Event Receiver reads and parses logs from windows event log API to collect local events as you see on Windows Event Viewer.

Following configuration demonstrates:
 1. Collect : Collect Application, Security and System channels 
 2. Transform : Set \_sourceCatetgory field to "windows_event_log_prod" 

>  3. Export : Send data to authenticated Sumologic organization

Reviewer 2: See https://github.com/SumoLogic/sumologic-documentation/pull/2501/files#r1208088856 for formatting

> Following configuration demonstrates:
>  1. Collect : Collect Application, Security and System channels 
>  2. Transform : Set \_sourceCatetgory field to "windows_event_log_prod" 
>  3. Export : Send data to authenticated Sumologic organization
>  

Reviewer 2: ```suggestion
Following configuration demonstrates:
1. Collect. Collect Application, Security and System channels. 
2. Transform. Set `_sourceCatetgory` field to "windows_event_log_prod".
3. Export. Send data to authenticated Sumo Logic organization.
```

 
```yaml
receivers:
  windowseventlog/application/localhost:
    channel: Application
  windowseventlog/security/localhost:
    channel: Security
  windowseventlog/system/localhost:
    channel: System

processors:
  resource/windows_resource_attributes/localhost:
    attributes:
      - key: _sourceCategory
        value: windows_event_log_prod
        action: insert
service:
  pipelines:
    logs/windows/localhost:
      receivers:
        - windowseventlog/application/localhost
        - windowseventlog/system/localhost
        - windowseventlog/security/localhost
      processors:
        - resource/windows_resource_attributes/localhost
        - batch
        - memory_limiter
      exporters:
        - sumologic
```
### Configuration details

>  1. Create a file in folder /etc/otelcol-sumo/conf.d  with name sample_windows.yaml.
>  2. Paste the above content.
>  3. Restart collector with following command:
>       - Restart-Service -Name OtelcolSumo

Reviewer 2: ```suggestion
1. Create a file in folder /etc/otelcol-sumo/conf.d  with name sample_windows.yaml.
2. Paste the above content.
3. Restart collector with following command: `Restart-Service -Name OtelcolSumo`.
```

 
* __exporters__: 
  *  `sumologic` Sends data to registered Sumologic Organization.  Auto configured in sumologic.yaml during installation using script.
 
* __service__: 
  *  `logs/windows/localhost:`  Pipeline glues together what data to collect(receiver), how to process the data(processors) and where to send logs(exporter). 

For more details, see the [Windows Event Log receiver][windowseventlogreceiver].

## Syslog Receiver

The OpenTelemetry Collector offers two approaches for Syslog processing:

* [Syslog Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/syslogreceiver)
* [TCPlog](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/tcplogreceiver) / [UDPlog Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/udplogreceiver)

Following configuration demonstrates:
 1. Collect : Collect syslog sent using udp protocol to 127.0.0.1 on port 514
 2. Transform : Set \_sourceCatetgory field to "syslog_event_log_prod" 
 3. Export : Send data to authenticated Sumologic organization

```yaml

receivers:
  syslog:

>      udp:

Reviewer 3: ```suggestion
    udp:
```

>     listen_address: "127.0.0.1:514"

Reviewer 3: ```suggestion
      listen_address: "127.0.0.1:514"
```

   

processors:
  resource/syslog:
    attributes:
      - key: _sourceCategory
        value: syslog_event_log_prod
        action: insert

 
service:
  pipelines:
    logs/syslog:
      receivers: 
        - syslog
      processors: 

>         - resource/syslog:

Reviewer 3: ```suggestion
        - resource/syslog
```

      exporters: 

>               sumologic

Reviewer 3: ```suggestion
        - sumologic
```

  ```

### Configuration details

>  1. Create a file in folder /etc/otelcol-sumo/conf.d  with name for your choice, sample_syslog_configuration.yaml.
>  2. Paste the above content.
>  3. Restart collector with following command:
>       - Linux : systemctl restart otelcol-sumo

Reviewer 2: ```suggestion
1. Create a file in folder /etc/otelcol-sumo/conf.d  with name for your choice, sample_syslog_configuration.yaml.
2. Paste the above content.
3. Restart collector with following command:
      - Linux: `systemctl restart otelcol-sumo`
```

 2. Paste the above content.
 3. Restart collector with following command:
   ```bash title="Linux"
   systemctl restart otelcol-sumo
   ```
   ```bash title="Windows"
   Restart-Service -Name OtelcolSumo
   ```

* **receivers**:
  * `filelog/custom_files:` Unique name defined for the filelog receiver. You can add multiple file receivers as filelog/2 , filelog/prod etc.
  * `include:` Lets the Filelog Receiver know where the log file is placed. Make sure that the collector has permissions to access the files; otherwise, it will not be collected.
  * `include_file_path_resolved: true` Adds a `log.file.path_resolved` attribute to the logs that contain the whole path of the file.
  * `storage: file_storage:` Prevents the receiver from reading the same data over and over again on each otelcol restart. Extension defined by default in sumologi.yaml
* **processors**:
  * `resource/1` Name of the processor that uses `attributes` to add `_sourceCategory` field with value `application_logs_prod`.
* **exporters**:
  * `sumologic` Sends data to registered Sumologic Organization. Auto-configured in sumologic.yaml during installation using script.
* **service**:

>   *  `logs/pipeline1:` Pipeline glues together what data to collect(receiver), how to process the data(processors) and where to send logs(exporter).

Reviewer 2: ```suggestion
  * `logs/pipeline1:` Pipeline glues together what data to collect(receiver), how to process the data(processors) and where to send logs(exporter).
```

      - sumologic
```

> ## Windows Log Event Receiver

Reviewer 3: Why is the section "Windows Log Event Receiver" removed?

Reviewer 3: Oh, sorry, missed it - the section is just moved up.
